doc/source/whatsnew/v1.0.0.rst

```diff
@@ -292,6 +292,34 @@ New repr for :class:`~pandas.arrays.IntervalArray`
 
    pd.arrays.IntervalArray.from_tuples([(0, 1), (2, 3)])
 
+- :meth:`Dataframe.info` now shows line numbers for the columns summary (:issue:`17304`)
```

Comment:
can you add a title and underline here (like other sub-sections).

Reply:
Added an underlined title for the new feature, thanks for the feedback